core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -238,12 +239,13 @@ public RewriteDataFilesActionResult execute() {
     }
 
     List<DataFile> addedDataFiles = rewriteDataForTasks(combinedScanTasks);
-    List<DataFile> currentDataFiles = combinedScanTasks.stream()
+    Set<DataFile> currentDataFiles = combinedScanTasks.stream()
```

Comment:
Why is there a change in this bit of code? I see we are creating a set but why would we have duplicates?

Reply:
If we split a large file into multiple small files, the original large file will appear multiple times in `currentDataFiles`, so I use the `set` to remove duplicates